yt_dlp/extractor/reddit.py

```diff
@@ -326,6 +370,8 @@ def add_thumbnail(src):
                 dash_playlist_url, display_id, mpd_id='dash', fatal=False)
             formats.extend(dash_fmts)
             self._merge_subtitles(dash_subs, target=subtitles)
+            if not subtitles and self._is_valid_url(caption_url, video_id, item='subtitle'):
```

Comment:
`caption_url` can go here
```suggestion
            caption_url = f'https://v.redd.it/{video_id}/wh_ben_en.vtt'
            if not subtitles and self._is_valid_url(caption_url, video_id, item='subtitle'):
```